pkgs/development/tools/yarn2nix-moretea/yarn2nix/lib/generateNix.js

```diff
@@ -46,37 +46,55 @@ function prefetchgit(url, rev) {
 }
 
 function fetchgit(fileName, url, rev, branch, builtinFetchGit) {
+  const repo = builtinFetchGit
+    ? `builtins.fetchGit ({
+         url = "${url}";
+         ref = "${branch}";
+         rev = "${rev}";
+       } // (if builtins.compareVersions "2.4pre" builtins.nixVersion < 0 then {
+         # workaround for https://github.com/NixOS/nix/issues/5128
+         allRefs = true;
+       } else {}))`
+    : `fetchgit {
+         url = "${url}";
+         rev = "${rev}";
+         sha256 = "${prefetchgit(url, rev)}";
+       }`;
+
   return `    {
     name = "${fileName}";
     path =
-      let${
-        builtinFetchGit
-          ? `
-        repo = builtins.fetchGit ({
-          url = "${url}";
-          ref = "${branch}";
-          rev = "${rev}";
-        } // (if builtins.compareVersions "2.4pre" builtins.nixVersion < 0 then {
-          # workaround for https://github.com/NixOS/nix/issues/5128
-          allRefs = true;
-        } else {}));
-      `
-          : `
-        repo = fetchgit {
-          url = "${url}";
-          rev = "${rev}";
-          sha256 = "${prefetchgit(url, rev)}";
-        };
-      `
-      }in
-        runCommand "${fileName}" { buildInputs = [gnutar]; } ''
-          # Set u+w because tar-fs can't unpack archives with read-only dirs
-          # https://github.com/mafintosh/tar-fs/issues/79
-          tar cf $out --mode u+w -C \${repo} .
-        '';
+      let repo = ${repo};
+      in runCommand "${fileName}" { buildInputs = [gnutar]; } ''
+        # Set u+w because tar-fs can't unpack archives with read-only dirs
+        # https://github.com/mafintosh/tar-fs/issues/79
+        tar cf $out --mode u+w -C \${repo} .
+      '';
   }`;
 }
 
+/**
+ * Parse an integrity hash out of an SSRI string.
+ *
+ * Provides a default and uses the "best" supported algorithm if there are multiple.
+ */
+function parseIntegrity(maybeIntegrity, fallbackHash) {
+  if (!maybeIntegrity && fallbackHash) {
+    return { algo: "sha1", hash: fallbackHash };
+  }
+
+  const integrities = ssri.parse(maybeIntegrity);
+  for (const key in integrities) {
+    if (!/^sha(1|256|512)$/.test(key)) {
+      delete integrities[key];
+    }
+  }
+
+  algo = integrities.pickAlgorithm();
+  hash = integrities[algo][0].digest;
```

Comment:
Weren't we supposed to use `hexDigest()` ?

Reply:
previously we were using hex digests because that's what is in the integrity line on the lock file, and the only processing we would do on that is splitting the digest and Algo.

however, since SSRI already parses the digests into base64, [jtojnar suggested](https://github.com/NixOS/nixpkgs/pull/197185#discussion_r1006475038) suggested we just use that formay, since nix supports it. base64 is much more compact than base16, so it will save a little disk space for large yarn projects